src/mito2/src/worker/handle_rebuild_index.rs

```diff
@@ -0,0 +1,205 @@
+
+        let puffin_manager = if let Some(write_cache) = self.cache_manager.write_cache() {
+            write_cache.build_puffin_manager()
+        } else {
+            access_layer.build_puffin_manager()
+        };
+
+        let intermediate_manager = if let Some(write_cache) = self.cache_manager.write_cache() {
+            write_cache.intermediate_manager().clone()
+        } else {
+            access_layer.intermediate_manager().clone()
+        };
+
+        let indexer_builder_ref = Arc::new(IndexerBuilderImpl {
+            op_type: OperationType::Flush, //TODO(SNC123): Temporarily set to Flush, 4 BuildTypes will be introduced later.
+            metadata: version.metadata.clone(),
+            inverted_index_config: self.config.inverted_index.clone(),
+            fulltext_index_config: self.config.fulltext_index.clone(),
+            bloom_filter_index_config: self.config.bloom_filter_index.clone(),
+            index_options: version.options.index_options.clone(),
+            row_group_size: WriteOptions::default().row_group_size,
+            intermediate_manager,
+            puffin_manager,
+        });
+
+        IndexBuildTask {
+            file_meta: file.meta_ref().clone(),
+            reason: build_type,
+            flushed_entry_id: Some(version.flushed_entry_id),
+            flushed_sequence: Some(version.flushed_sequence),
+            access_layer: access_layer.clone(),
+            write_cache: self.cache_manager.write_cache().cloned(),
+            file_purger: file.file_purger(),
+            request_sender: self.sender.clone(),
+            indexer_builder: indexer_builder_ref.clone(),
+            result_sender,
+        }
+    }
+
+    pub(crate) async fn handle_rebuild_index(&mut self, request: RegionBuildIndexRequest) {
+        let region_id = request.region_id;
+        let Some(region) = self.regions.get_region(region_id) else {
+            return;
+        };
+
+        let version = region.version();
+
+        let all_files: HashMap<FileId, FileHandle> = version
+            .ssts
+            .levels()
+            .iter()
+            .flat_map(|level| level.files.iter())
+            .filter(|(_, handle)| !handle.is_deleted())
+            .map(|(id, handle)| (*id, handle.clone()))
+            .collect();
+
+        let build_tasks = if request.file_metas.is_empty() {
+            all_files.values().cloned().collect::<Vec<_>>()
+        } else {
+            request
+                .file_metas
+                .iter()
+                .filter_map(|meta| all_files.get(&meta.file_id).cloned())
+                .collect::<Vec<_>>()
+        };
+
+        for file_handle in build_tasks {
+            let _ = self
+                .index_build_scheduler
+                .schedule_build(self.new_index_build_task(
+                    &region,
+                    file_handle,
+                    request.build_type.clone(),
+                    None,
+                ));
+        }
```

Comment:
A prominent note is needed: Currently, rebuilding the index will reconstruct the index for all files in the region, which is a simplified approach and is not yet available for production use; further optimization is required.